taskcollector/TaskCollector.cpp

```diff
@@ -1,81 +1,78 @@
 #include "FreeImage.h"
+#include <algorithm>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <list>
+#include <memory>
 #include <queue>
 #include <iostream>
 
-
-std::queue<FIBITMAP *> alphaChunks;	
+using bitmap_t = std::shared_ptr<FIBITMAP>;
 
 /** Generic image loader
-	@param lpszPathName Pointer to the full file name
-	@param flag Optional load flag constant
-	@return Returns the loaded dib if successful, returns NULL otherwise
+@param lpszPathName Pointer to the full file name
+@param flag Optional load flag constant
+@return Returns the loaded dib if successful, returns NULL otherwise
 */
-FIBITMAP* GenericLoader(const char* lpszPathName, int flag) {
+bitmap_t GenericLoader(const char* lpszPathName, int flag) {
 	FREE_IMAGE_FORMAT fif = FIF_UNKNOWN;
 
 	// check the file signature and deduce its format
 	// (the second argument is currently not used by FreeImage)
 	fif = FreeImage_GetFileType(lpszPathName, 0);
-	if(fif == FIF_UNKNOWN) {
+	if (fif == FIF_UNKNOWN) {
 		// no signature ?
 		// try to guess the file format from the file extension
 		fif = FreeImage_GetFIFFromFilename(lpszPathName);
 	}
 	// check that the plugin has reading capabilities ...
-	if((fif != FIF_UNKNOWN) && FreeImage_FIFSupportsReading(fif)) {
+	if ((fif != FIF_UNKNOWN) && FreeImage_FIFSupportsReading(fif)) {
 		// ok, let's load the file
-		FIBITMAP *dib = FreeImage_Load(fif, lpszPathName, flag);
+		auto dib = std::make_shared<FIBITMAP>(*FreeImage_Load(fif, lpszPathName, flag));
 		// unless a bad file format, we are done !
 		return dib;
 	}
 	return NULL;
 }
 
-unsigned int min(unsigned int a, unsigned int b) {
-	return a < b ? a : b;
-}
-
 /** Generic image writer
-	@param dib Pointer to the dib to be saved
-	@param lpszPathName Pointer to the full file name
-	@param flag Optional save flag constant
-	@return Returns true if successful, returns false otherwise
+@param dib Pointer to the dib to be saved
+@param lpszPathName Pointer to the full file name
+@param flag Optional save flag constant
+@return Returns true if successful, returns false otherwise
 */
-bool GenericWriter(FIBITMAP* dib, const char* lpszPathName, int flag) {
+bool GenericWriter(bitmap_t dib, const char* lpszPathName, int flag) {

```

Comment:
There is no need to pass a copy of shared pointer to the function. The function need not to be the owner of the bitmap.